extensions/data-plane/data-plane-google-bigquery/src/main/java/org/eclipse/edc/connector/dataplane/gcp/bigquery/pipeline/BigQueryDataSourceFactory.java

```diff
@@ -0,0 +1,104 @@
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LLC
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.gcp.bigquery.pipeline;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import org.eclipse.edc.connector.dataplane.gcp.bigquery.params.BigQueryRequestParamsProvider;
+import org.eclipse.edc.connector.dataplane.spi.pipeline.DataSource;
+import org.eclipse.edc.connector.dataplane.spi.pipeline.DataSourceFactory;
+import org.eclipse.edc.gcp.bigquery.BigQueryConfiguration;
+import org.eclipse.edc.gcp.bigquery.service.BigQueryServiceSchema;
+import org.eclipse.edc.gcp.common.GcpException;
+import org.eclipse.edc.gcp.iam.IamService;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.eclipse.edc.spi.types.domain.transfer.DataFlowStartMessage;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.ExecutorService;
+
+import static org.eclipse.edc.gcp.bigquery.service.BigQueryServiceSchema.BIGQUERY_DATA;
+
+/**
+ * Instantiates the BigQuery data transfer source {@link BigQueryDataSource} for requests whose
+ * source data type is BigQueryData. This factory is registered by the BigQuery Data Plane Extension.
+ */
+public class BigQueryDataSourceFactory implements DataSourceFactory {
+    private final BigQueryConfiguration configuration;
+    private final BigQueryRequestParamsProvider requestParamsProvider;
+    private final Monitor monitor;
+    private final TypeManager typeManager;
+    private final ExecutorService executorService;
+    private IamService iamService;
+
+    public BigQueryDataSourceFactory(BigQueryConfiguration configuration, Monitor monitor,
+                                     BigQueryRequestParamsProvider requestParamsProvider,
+                                     TypeManager typeManager, ExecutorService executorService,
+                                     IamService iamService) {
+        this.configuration = configuration;
+        this.monitor = monitor;
+        this.requestParamsProvider = requestParamsProvider;
+        this.typeManager = typeManager;
+        this.executorService = executorService;
+        this.iamService = iamService;
+    }
+
+    @Override
+    public boolean canHandle(DataFlowStartMessage message) {
+        return BigQueryServiceSchema.BIGQUERY_DATA.equals(message.getSourceDataAddress().getType());
+    }
+
+    @Override
+    public @NotNull Result<Void> validateRequest(DataFlowStartMessage message) {
+        // canHandle has been already invoked to have this factory selected.
+        // BigQuerySinkDataAddressValidator has already checked message.getDestinationDataAddress().
+        // BigQuerySourceDataAddressValidator has already checked message.getSourceDataAddress().
+
+        return Result.success();
```

Comment:
This is the place where validators should be engaged.

Reply:
The extension [BigQueryDataAddressValidatorExtension](https://github.com/eclipse-edc/Technology-Gcp/blob/main/extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/bigquery/validation/BigQueryDataAddressValidatorExtension.java) is registering both source and sink validators: the validators are invoked by the transfer process service even before this method is invoked. This method is there to have the source factory implementing all methods of the interface.

Reply:
but this is the dataplane. Generally, you can't trust your communication partner to do stuff correctly, and here in particular, because `DataFlowStartMessages` could be sent over Dataplane Signaling by a component that is _not_ an EDC controlplane.

Reply:
Will check in createSource as written below